integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,259 @@
+    String alluxioPath = fuseCliOpts.getMountAlluxioPath().orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH));
+    Optional<List<String>> libfuseOptionsFromCli = fuseCliOpts.getFuseOptions();
+    List<String> libfuseOptions;
+    if (libfuseOptionsFromCli.isPresent()) {
+      libfuseOptions = libfuseOptionsFromCli.get();
+    } else {
+      if (conf.isSet(PropertyKey.FUSE_MOUNT_OPTIONS)) {
+        libfuseOptions = conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS);
+      } else {
+        libfuseOptions = ImmutableList.of();
+      }
+    }
+    libfuseOptions = optimizeAndFormatFuseOptions(libfuseOptions);
+    String mountPoint = fuseCliOpts.getMountPoint().orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_POINT));
+    Optional<String> authPolicyCustomGroup = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP)
+        && !conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP).isEmpty()
+        ? Optional.of(conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP))
+        : Optional.empty();
+    Optional<String> authPolicyCustomUser = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER)
+        && !conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER).isEmpty()
+        ? Optional.of(conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER))
+        : Optional.empty();
+    return new AlluxioFuseFileSystemOpts(alluxioPath,
+        conf.getString(PropertyKey.FUSE_FS_NAME),
+        conf.getClass(PropertyKey.FUSE_AUTH_POLICY_CLASS),
+        authPolicyCustomGroup,
+        authPolicyCustomUser,
+        conf.getInt(PropertyKey.FUSE_CACHED_PATHS_MAX),
+        (int) conf.getMs(PropertyKey.FUSE_UMOUNT_TIMEOUT),
+        conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED),
+        libfuseOptions,
+        conf.getBoolean(PropertyKey.USER_METADATA_CACHE_ENABLED),
+        mountPoint,
+        conf.getBoolean(PropertyKey.FUSE_SPECIAL_COMMAND_ENABLED),
+        conf.getMs(PropertyKey.FUSE_STAT_CACHE_REFRESH_INTERVAL),
+        conf.getBoolean(PropertyKey.FUSE_USER_GROUP_TRANSLATION_ENABLED));
+  }
+
+  private AlluxioFuseFileSystemOpts(String alluxioPath, String fsName, Class fuseAuthPolicyClass,
+        Optional<String> fuseAuthPolicyCustomGroup, Optional<String> fuseAuthPolicyCustomUser,
+        int fuseMaxPathCached, int fuseUmountTimeout, boolean isDebug, List<String> libfuseOptions,
+        boolean metaDataCacheEnabled, String mountPoint, boolean specialCommandEnabled,
+        long statCacheTimeout, boolean userGroupTranslationEnabled) {
+    mAlluxioPath = Preconditions.checkNotNull(alluxioPath,
+        "Option alluxioPath for an AlluxioFuse filesystem should not be null.");
+    mFsName = Preconditions.checkNotNull(fsName,
+        "Option fsName for an AlluxioFuse filesystem should not be null.");
+    mFuseAuthPolicyClass = Preconditions.checkNotNull(fuseAuthPolicyClass,
+        "Option fuseAuthPolicyClass for an AlluxioFuse filesystem should not be null.");
```

Comment:
i guess the should not be null message does not include new information than the one directly throw by checkNotNull?

Reply:
i mean the check not null message do not show new information that we can avoid adding that message
directly 
Preconditions.checkNotNull(variable)